### PR TITLE
Resolve matplotlib deprecation issue

### PR DESCRIPTION
**Pull request by afni-dglen**
Closes #
https://github.com/physiopy/peakdet/pull/66#issuecomment-2232646353

Couldn't get rebase to work, so reforked again and put the changes for editor.py and setup.cfg requested in that previous issue.

## Proposed Changes
Implement changes for deprecated rectprops with props in matplotlib function call. This version checks for older versions of matplotlib to continue with old behavior.
Also removed some of the constraints on matplotlib in setup.cfg, but kept requirement for some version, consistent with physiobids setup.cfg.

## Change Type

- [ ] `bugfix` (+0.0.1)
- [x] `minor` (+0.1.0)
- [ ] `major`  (+1.0.0)
- [ ] `refactoring` (no version update)
- [ ] `test` (no version update)
- [ ] `infrastructure` (no version update)
- [ ] `documentation` (no version update)
- [ ] `other`

## Checklist before review

- [ ] I added everything I wanted to add to this PR.
- [ ] \[Code or tests only\] I wrote/updated the necessary docstrings.
- [ ] \[Code or tests only\] I ran and passed tests locally.
- [ ] \[Documentation only\] I built the docs locally.
- [ ] My contribution is harmonious with the rest of the code: I'm not introducing repetitions.
- [ ] My code respects the adopted style, especially linting conventions.
- [ ] The title of this PR is explanatory on its own, enough to be understood as part of a changelog.
- [ ] I added or indicated the right labels.

- [ ] I added information regarding the timeline of completion for this PR.

- [ ] Please, comment on my PR while it's a draft and give me feedback on the development!

### CODE REVIEW

File: peakdet/editor.py [hunk 1]
```diff
@@ -60,29 +60,36 @@ def __init__(self, data):
         delete = functools.partial(self.on_edit, method="delete")
         reject = functools.partial(self.on_edit, method="reject")
         insert = functools.partial(self.on_edit, method="insert")
+
+        # Check matplotlib version rectprops is deprecated with matplotlib 3.5.0 and then obsolete 
+        if matplotlib.__version__ >= '3.5.0':
```
Comment (maestroque) [inline peakdet/editor.py:65]: You need to explicitly `import matplotlib` for this. The current imports don't cover this. This is also indicated in the `pre-commit` fail

**Review by maestroque — COMMENTED** [review]
Currently, the `pre-commit` CI test is failing. You should have this installed in your local repository first, so that the pre-commit hooks run locally, only allowing to commit once the checks pass.
```bash
# Given the virtual environment you are working on is activated
pip install pre-commit
pre-commit install
# Now, each time you commit the pre-commit hooks will run only allowing for the commit once all of the tests pass

# Run the following command to run the test prior to commiting for a sanity check
pre-commit run --all-files
```

Now you would need to run the pre-commit hooks to correct some styling issues and then re-add the files and commit. Otherwise you will get a warning for unstaged files altered by pre-commit

@afni-dglen

**Review by maestroque — APPROVED** [review]
Looks good to me 🚀 

Thanks @afni-dglen

**Comment by smoia** [comment]
:rocket: PR was released in [`0.5.0`](https://github.com/physiopy/peakdet/releases/tag/0.5.0) :rocket:

**Pull request merged**